src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs

```diff
@@ -62,7 +129,7 @@ public static IServiceCollection AddMvc([NotNull] this IServiceCollection servic
 
             // This provider needs access to the per-request services, but might be used many times for a given
             // request.
-            services.TryAdd(ServiceDescriptor.Transient<IActionConstraintProvider, DefaultActionConstraintProvider>());
+            services.AddTransient<IActionConstraintProvider, DefaultActionConstraintProvider>();
```

Comment:
When one of the newly-multiple providers (`IActionConstraintProvider`, `IActionDescriptorProvider`, `IActionInvokerProvider`, `IApiDescriptionProvider`, `IFilterProvider`) has multiple _identical_ registrations, what can go South?

For example looks like `ControllerActionDescriptorProvider` and `DefaultApiDescriptionProvider` will duplicate descriptors in their `context.Results` collections. Is that alright?